### PR TITLE
add c-api for allowing FIFO compaction

### PR DESCRIPTION
**Pull request by hackingthekernel**
Addressing issue #11079

### CODE REVIEW

**Review by ajkr — APPROVED** [review]
LGTM, thank you!

File: db/c.cc [hunk 1]
```diff
@@ -5072,6 +5072,16 @@ rocksdb_fifo_compaction_options_t* rocksdb_fifo_compaction_options_create() {
   return result;
 }
 
+void rocksdb_fifo_compaction_options_set_allow_compaction(
+    rocksdb_fifo_compaction_options_t* fifo_opts, bool allow_compaction) {
```
Comment (ajkr) [inline db/c.cc:5076]: Although `bool` works as far as I can tell, almost everywhere else uses `unsigned char` in this file and I found it's documented in the header - https://github.com/facebook/rocksdb/blob/24ac53d81a2c97f6c30305bd2c3a12f0800ab945/include/rocksdb/c.h#L40. Do you mind changing to that for consistency?

**Review by ajkr — CHANGES_REQUESTED** [review]
Actually I have one hopefully small request

File: db/c.cc (hunk 1 above)
Comment (hackingthekernel) [reply db/c.cc]: Indeed! Thank you!

**Pull request closed without merging**